### DISCUSSION TITLE
How to get the position of the vehicle (geographical coordinates of the vehicle) in MOSAIC during the simulation.

### DISCUSSION THREAD

**Discussion by Jean-Tshibangu-jtm** [eclipse/mosaic#393] [Q&A]
How **to get the position of the vehicle (geographical coordinates of the vehicle)** in MOSAIC during the simulation.

My idea is to check whether the geographical position of the vehicle belongs to a geographical zone, in order to trigger an event.

I'm sending you a few lines of code that I've written, but on the SUMO side there are errors (including roads).

 **GeoPoint currentPosition = getOs().getPosition();**

I used this MOSAIC interface :  import org.eclipse.mosaic.lib.objects.road.IRoadPosition;

![ErrorTrace](https://github.com/eclipse/mosaic/assets/50569553/fea53d84-cd39-4536-bff3-4cd733c5fa8c)

 
Thank you for your help.

**Comment by schwepmo** [comment]
Hello @Jean-Tshibangu-jtm,
The `getOs().getPosition()` function should do exactly what you've described, though you have to be careful that the traffic simulator (i.e., SUMO) is already providing the respective information. Most of the time this means that you query `getOs().getPosition()` only after the first time `onVehicleUpdated()` has been called (e.g., by adding an `if (previousVehicleDate == null) { ... }` at the top).

The error that SUMO is printing relates to something different. The command at question is relating to a state change (https://sumo.dlr.de/docs/TraCI/Change_Vehicle_State.html), e.g., when you call `changeLane(...)` or similar methods. 

I'm assuming that you're using a flows within a SUMO route-file to spawn the vehicles. But to further assist you I would need to know more about the scenario and applications.

So to summarize `getOs().getPosition()` should work and shouldn't correlate with the SUMO errors. The SUMO errors have a different cause, e.g., trying to change the lanes of vehicles that have already despawned.

Let me know if these hints help you or provide some further information on your setup.
